index.dd

```diff
@@ -53,6 +53,28 @@ void main()
 )
 )
 $(EXTRA_EXAMPLE
+----
+void main()
+{
+    import std.stdio : writefln;
+    int[char[2]] aa;
+
+    // The string `arr` has a limited alphabet: {A, C, G, T}
+    // Thus, iteration can be done _without_ decoding
+    auto arr = "AGATAGA";
+
+    // iterate over all pairs in the string and observe each pair
+    // ('A', 'G'), ('G', 'A'), ('A', 'T'), ...
+    foreach (i; 0 .. arr.length - 1)
+        aa[arr[i .. $][0 .. 2]]++;
```

Comment:
Might be worth mentioning/implying that slicing a string does not allocate a copy